packages/core/src/odata-common/batch-response.ts

```diff
@@ -17,7 +17,7 @@ export interface ReadResponse {
   httpCode: number;
   body: Record<string, any>;
   type: Constructable<EntityBase>;
-  as: <T extends EntityBase>(constructor: Constructable<T>) => Error | T[];
+  as: <T extends EntityBase>(constructor: Constructable<T>) => T[];
```

Comment:
If I remember correctly, the `Error | T[]` type is there from the beginning for some reasons.
In this way, at least release notes and the compatibility should be touched.